src/services/ledger/paginated-ledger.ts

```diff
@@ -53,4 +52,13 @@ export const paginatedLedger = async (
       session: options.session,
     })
     .toArray()
+
+  const countPromise = Transaction.collection.countDocuments(filterQuery, {
+    session: options.session,
+  })
+
+  return {
+    slice: await findPromise,
+    total: await countPromise,
+  } as PaginatedArray<ILedgerTransaction>
```

Comment:
can we avoid the casting? having `findPromise` and `countPromise` return the right type?